src/python/grpcio/grpc/_adapter/_low.py

```diff
@@ -170,10 +170,14 @@ def set_credentials(self, creds):
 class Channel(_types.Channel):
 
   def __init__(self, target, args, creds=None):
-    args = list(args) + [
-        (cygrpc.ChannelArgKey.primary_user_agent_string, _USER_AGENT)]
+    args = dict(args)
```

Comment:
Actually the opposite... at least the user agent strings are already
allowed to be repeated. There might be more.

On Wed, Jun 8, 2016 at 8:04 PM Masood Malekghassemi <
notifications@github.com> wrote:

> In src/python/grpcio/grpc/_adapter/_low.py
> <https://github.com/grpc/grpc/pull/6764#discussion_r66376373>:
>
> > @@ -170,10 +170,14 @@ def set_credentials(self, creds):
> >  class Channel(_types.Channel):
> >
> >    def __init__(self, target, args, creds=None):
> > -    args = list(args) + [
> > -        (cygrpc.ChannelArgKey.primary_user_agent_string, _USER_AGENT)]
> > +    args = dict(args)
>
> @ctiller <https://github.com/ctiller>: is it guaranteed that duplicated
> channel argument keys will never be useful (e.g. channel arguments are
> never a multimap)?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/grpc/grpc/pull/6764/files/bf4776a7b61be4eeda73805098a8011131f7557a#r66376373>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AJpudW4y5DFCpnyRrUiAJ3y64vfoReAqks5qJ4KlgaJpZM4Is9r2>
> .
>